### PR TITLE
refactor(npm): lockfile extraction code

### PR DESCRIPTION
**Pull request by ylemkimon**
## Changes:
Refactored lockfile extraction code.

## Context:
For #7700

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/npm/extract/locked-versions.spec.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ describe('manager/npm/extract/locked-versions', () => {
       async (yarnVersion) => {
         yarn.getYarnLock.mockReturnValue({
           isYarn1: yarnVersion === '1.22.0',
-          cacheVersion: yarnVersion === '2.2.0' ? 6 : NaN,
+          lockfileVersion: yarnVersion === '2.2.0' ? 6 : undefined,
```
Comment (viceice) [inline lib/manager/npm/extract/locked-versions.spec.ts:18]: Why `undefined` instead of `NaN` ?

**Review by viceice — COMMENTED** [review]
Just one Question: Why switch to `undefined` instead of existing `NaN` ? Otherwise LGTM

File: lib/manager/npm/extract/locked-versions.spec.ts (hunk 1 above)
Comment (ylemkimon) [reply lib/manager/npm/extract/locked-versions.spec.ts:18]: @viceice I wasn't sure `undefined` is converted to `NaN` in integer comparisons and worried about TypeScript errors in #7204. Now I think it seems that implicit `undefined` is cleaner than explicit `NaN`.

Comment (viceice) [reply lib/manager/npm/extract/locked-versions.spec.ts:18]: ok

File: lib/manager/npm/extract/yarn.ts [hunk 2]
```diff
@@ -2,42 +2,37 @@ import { structUtils } from '@yarnpkg/core';
...
-  lockedVersions: Record<string, string>;
-}> {
+export async function getYarnLock(filePath: string): Promise<LockFile> {
   const yarnLockRaw = await readLocalFile(filePath, 'utf8');
   try {
     const parsed = parseSyml(yarnLockRaw);
-    const lockFile: Record<string, string> = {};
-    let cacheVersion = NaN;
+    const lockedVersions: Record<string, string> = {};
+    let lockfileVersion;
```
Comment (viceice) [inline lib/manager/npm/extract/yarn.ts:12]:
```suggestion
    let lockfileVersion: number;
```

**Review by viceice — COMMENTED** [review]
tested `NaN` vs `undefined`

```ts
undefined < 6 === false
undefined > 6 === false
NaN > 6 === false
NaN < 6 === false
```

just a small type change

File: lib/manager/npm/extract/yarn.ts (hunk 2 above)
Comment (ylemkimon) [reply lib/manager/npm/extract/yarn.ts]: Done :smile:

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**